### PR TITLE
Feat: refactor CLI commands related to resources

### LINKED ISSUE #4485: The `vela exec` command can not find the pod generated by HelmRelease
**Describe the bug**

The `vela exec` command can not find the pod generated by HelmRelease

```bash
root@cloudshell-users-admin--1-mtslh:/# vela exec addon-dex -n vela-system -- bash
Error: no supported resources detected in deployed resources
```

**To Reproduce**

1. Deploy an app with the helm-type component, such as the `dex` addon.
2. Execute `vela exec`

**Expected behavior**

Provide the pod list to be selected.

**Screenshots**

**KubeVela Version**

1.5.0-beta.4

### LINKED ISSUE #4488: `vela port-forward` provides the duplicate options.
**Describe the bug**

image

**To Reproduce**

```
vela port-forward -n vela-system addon-dex
```

**KubeVela Version**

1.5.0-beta.3

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

#### VelaQL

* Add a new provider: `ql.#CollectServices`

>  Support to query the service resources managed by the application. The data source is the application resource tree. 

* Refactor a provider: `ql.#CollectPods`

> Support to query the pod resources managed by the application. The data source is the application resource tree. 

#### CLI Command

* `vela exec`

```bash
vela exec appName -- bash
vela exec appName -n ns -- bash
vela exec appName -n ns -c componentName  -- bash
vela exec appName -n ns -c componentName --cluster local -- bash
vela exec appName -n ns -c componentName --cluster local --pod podName -- bash
vela exec appName -n ns -c componentName --cluster local --pod podName --container containerName  -- bash
```

* `vela port-forward`

```bash
vela port-forward APP_NAME
vela port-forward APP_NAME -n ns -c componentName
vela port-forward APP_NAME -n ns -c componentName --cluster local
vela port-forward APP_NAME -n ns -c componentName --cluster local --resource-name servicename 8080:80
vela port-forward APP_NAME -n ns -c componentName --cluster local --resource-type pod 8080:80
```

* `vela status --pod` (New)

```bash
$ vela status addon-velaux --pod -n vela-system 
CLUSTER COMPONENT       POD NAME                        NAMESPACE       PHASE   CREATE TIME             REVISION        HOST              
local   apiserver       apiserver-5c95478ddd-spc8d      vela-system     Running 2022-07-28T05:59:45Z                    test-control-plane
local   velaux          velaux-6bfb699994-gt457         vela-system     Running 2022-07-28T05:40:41Z                    test-control-plane
local   velaux          velaux-6bfb699994-t66gh         vela-system     Running 2022-07-28T14:04:03Z                    test-control-plane

$ vela status --pod addon-velaux -n vela-system --component componentName
$ vela status --pod addon-velaux -n vela-system --cluster local
```

* `vela logs`

```bash
vela logs appName
vela logs appName -n ns
vela logs appName -n ns -c componentName 
vela logs appName -n ns -c componentName --cluster local 
vela logs appName -n ns -c componentName --cluster local --pod podName
vela logs appName -n ns -c componentName --cluster local --pod podName --container containerName
```

Fixes #4485
Fixes #4488

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by wonderflow — CHANGES_REQUESTED** [review]
the `vela pods` command is confusing, let wrap it into 

```
vela status --pods
```

We already have:

```
vela status --tree
```

**Comment by barnettZQG** [comment]
> the `vela pods` command is confusing, let wrap it into
> 
> ```
> vela status --pods
> ```
> 
> We already have:
> 
> ```
> vela status --tree
> ```

fixed.

File: references/cli/portforward.go [hunk 1]
```diff
@@ -269,10 +272,7 @@ func (o *VelaPortForwardOptions) Complete() error {
 			}
 			return val
 		}
-		pt := o.targetResource.Endpoint.Port
-		if pt == 0 {
-			pt = getPortsFromApp(o.App)
-		}
+		pt := o.targetPort
```
Comment (wonderflow) [inline references/cli/portforward.go:275]: how do you get the targetPort if the resource is Pod

File: pkg/velaql/providers/query/collector.go [hunk 2]
```diff
@@ -492,35 +507,6 @@ func (c *HelmReleaseCollector) CollectIngress(ctx context.Context, cluster strin
 	return ingresses.Items, nil
 }
 
-// helmReleasePodCollector collect pods created by helmRelease
-func helmReleasePodCollector(cli client.Client, obj *unstructured.Unstructured, cluster string) ([]*unstructured.Unstructured, error) {
```
Comment (wonderflow) [inline pkg/velaql/providers/query/collector.go:496]: why not needed

**Review by wonderflow — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**